src/lib/lib.c

```diff
@@ -25,9 +25,22 @@ void *memset(void *s, int c, size_t n)
 {
 	uint8_t *d8 = s;
 	uint8_t v = c;
+	uint32_t v32 = (uint32_t)v | ((uint32_t)v << 8) |
+		((uint32_t)v << 16) | ((uint32_t)v << 24);
+	uint32_t *d32 = ALIGN_UP((uintptr_t)s, sizeof(uint32_t));
+	/* Don't bother with 32-bit copies for up to 7 bytes */
+	size_t prefix_sz = n > 2 * sizeof(v32) - 1 ? (uint8_t *)d32 - d8 : n;
 	int i;
 
-	for (i = 0; i <	n; i++)
+	for (i = 0; i < prefix_sz; i++)
+		d8[i] = v;
+
+	/* This won't be executed if n <= 7 */
+	for (i = 0; i < (n - prefix_sz) >> 2; i++)
+		d32[i] = v32;
+
+	/* This won't be executed if n <= 7 */
+	for (i = prefix_sz + (i << 2); i < n; i++)
```

Comment:
This line looks weird -- setting i from the old value of i. More comments explaining why this is correct are needed.